### PR TITLE
Stage1 of clang compiler warning patches.

### PR DESCRIPTION
**Pull request by ethoms**
Stage 1:

Fairly rudimentary fixes. Mostly fixing data type and casting inconsistencies.

Fixes for:
(i) Data type casting
(ii) Data type initialization
(iii) String formatting (data types)
(iv) Missing curly brackets in ambiguous if statements / for loops.

### CODE REVIEW

File: SoObjects/Contacts/SOGoContactSourceFolder.m [hunk 1]
```diff
@@ -94,7 +94,7 @@ - (id) initWithName: (NSString *) newName
     {
       if (![newDisplayName length])
         newDisplayName = newName;
-      ASSIGN (displayName, newDisplayName);
+      ASSIGN (displayName, [newDisplayName mutableCopy]);
```
Comment (extrafu) [inline SoObjects/Contacts/SOGoContactSourceFolder.m]: Why? You'll leak here.

File: SoObjects/Mailer/SOGoMailLabel.m [hunk 2]
```diff
@@ -74,7 +74,7 @@ + (NSArray *) labelsFromDefaults: (NSDictionary *) theDefaults
   int i;
 
   allLabels = [NSMutableArray array];
-  allKeys = [[theDefaults allKeys] sortedArrayUsingSelector: @selector (caseInsensitiveCompare:)];
+  allKeys = [[[theDefaults allKeys] sortedArrayUsingSelector: @selector (caseInsensitiveCompare:)] mutableCopy];
```
Comment (extrafu) [inline SoObjects/Mailer/SOGoMailLabel.m]: Also, why? It'll leak.

File: SoObjects/SOGo/SOGoFolder.m [hunk 3]
```diff
@@ -107,7 +107,7 @@ - (void) dealloc
 
 - (void) setDisplayName: (NSString *) newDisplayName
 {
-  ASSIGN (displayName, newDisplayName);
+  ASSIGN (displayName, [newDisplayName mutableCopy]);
```
Comment (extrafu) [inline SoObjects/SOGo/SOGoFolder.m]: Leaking...

File: SoObjects/SOGo/SOGoGCSFolder.m [hunk 4]
```diff
@@ -376,12 +376,12 @@ - (NSString *) displayName
   if (!displayName)
     {
       if (activeUserIsOwner)
-        displayName = [self _displayNameFromOwner];
+        displayName = [[self _displayNameFromOwner] mutableCopy];
```
Comment (extrafu) [inline SoObjects/SOGo/SOGoGCSFolder.m]: Leaking..

File: SoObjects/SOGo/SOGoGCSFolder.m [hunk 5]
```diff
@@ -376,12 +376,12 @@ - (NSString *) displayName
...
       else
         {
-          displayName = [self _displayNameFromSubscriber];
+          displayName = [[self _displayNameFromSubscriber] mutableCopy];
```
Comment (extrafu) [inline SoObjects/SOGo/SOGoGCSFolder.m]: Leaking..

File: SoObjects/SOGo/SOGoGCSFolder.m [hunk 6]
```diff
@@ -376,12 +376,12 @@ - (NSString *) displayName
...
           if (!displayName)
-            displayName = [self _displayNameFromOwner];
+            displayName = [[self _displayNameFromOwner] mutableCopy];
```
Comment (extrafu) [inline SoObjects/SOGo/SOGoGCSFolder.m]: Leaking..

File: UI/MailPartViewers/UIxMailPartViewer.m [hunk 7]
```diff
@@ -287,7 +287,7 @@ - (NSString *) _filenameForAttachment: (SOGoMailBodyPart *) bodyPart
   if ([filename length])
     // We replace any slash by a dash since Apache won't allow encoded slashes by default.
     // See http://httpd.apache.org/docs/2.2/mod/core.html#allowencodedslashes
-    filename = [filename stringByReplacingString: @"/" withString: @"-"];
+    filename = [[filename stringByReplacingString: @"/" withString: @"-"] mutableCopy];
```
Comment (extrafu) [inline UI/MailPartViewers/UIxMailPartViewer.m]: Leaking...

File: UI/Scheduler/UIxCalListingActions.m [hunk 8]
```diff
@@ -400,11 +400,11 @@ - (NSArray *) _fetchFields: (NSArray *) fields
           else if ([criteria isEqualToString:@"entireContent"])
             {
               // First search : Through the quick table inside the location, category and title columns
-              quickInfos = [currentFolder fetchCoreInfosFrom: startDate
+              quickInfos = [[currentFolder fetchCoreInfosFrom: startDate
                                                           to: endDate
                                                        title: value
                                                    component: component
-                                           additionalFilters: criteria];
+                                           additionalFilters: criteria] mutableCopy];
```
Comment (extrafu) [inline UI/Scheduler/UIxCalListingActions.m]: leaking

File: UI/Scheduler/UIxCalListingActions.m [hunk 9]
```diff
@@ -416,10 +416,10 @@ - (NSArray *) _fetchFields: (NSArray *) fields
                 }
         
               // Second research : Every objects except for those already in the quickInfos array
-              allInfos = [currentFolder fetchCoreInfosFrom: startDate
+              allInfos = [[currentFolder fetchCoreInfosFrom: startDate
                                                         to: endDate
                                                      title: nil
-                                                 component: component];
+                                                 component: component] mutableCopy];
```
Comment (extrafu) [inline UI/Scheduler/UIxCalListingActions.m]: leaking

**Comment by ethoms** [comment]
The reason I used mutableCopy is to (I thought) safely get a mutable instance. Since the compiler warns that it is the wrong data type being received. E.g. an NSMutableString is expected, but a NSString is given. Initially I just did a simple cast (NSMutableString *), but after reading that it's better to cast properly using mutableCopy I changed the patches.

Is it safe to use a simple cast (NSMutableString *) etc.? Or do have any better ideas to get the data types corrected so to avoid compiler warnings?

I'm not sure what you mean by leaking? I guess it should be passed by reference, not by copy?

**Comment by extrafu** [comment]
mutableCopy creates a new instance with a retain count of 1, so it's not autoreleased and thus, leaks the newly created instance.

Just cast it, it'll be good!

**Comment by ethoms** [comment]
OK, noted. So I'll just change these and add another commit to this PR?

**Comment by ethoms** [comment]
Actually, I'll just use the github editor. I'll post back after build testing on FreeBSD and RHEL 5.11.

**Comment by ethoms** [comment]
OK, all done. Converted all uses of mutableCopy to a cast in this pull request.

**Comment by ethoms** [comment]
NOTE: on line 168 of  UI/Scheduler/UIxCalDayView.m there is a use of mutableCopy. It's not one of my changes, but maybe you want to check it out.

**Comment by extrafu** [comment]
Yep, but notice it's returned as autoreleased at the end of the method.

**Comment by ethoms** [comment]
Ah, OK, I didn't read on.

**Comment by ethoms** [comment]
I've successfully build tested on both FreeBSD 10.1 and RHEL 5.11 after correcting the mutableCopy issue.

Is there any other issue with this PR?

**Comment by extrafu** [comment]
I don't think so but I'll wait until our nightly builds + integration tests pass in a couple of hours before merging new stuff.

Thanks!

**Comment by extrafu** [comment]
Can you please adapt this PR so we can merge it also in the v2 branch? I guess it'll be mostly code remove.

Open the new PR just for the v2 branch - since the patch was merged in master.

**Comment by ethoms** [comment]
No problem, I'll make a new PR to the v2 branch as soon as I've done looking into the other issues.

**Pull request merged**